packages/lesswrong/lib/collections/electionVotes/schema.ts

```diff
@@ -0,0 +1,51 @@
+import { foreignKeyField } from "../../utils/schemaUtils";
+import { userOwns } from "../../vulcan-users";
+
+const schema: SchemaType<DbElectionVote> = {
+  /** The name of the election */
+  electionName: {
+    type: String,
+    optional: false,
+    nullable: false,
+    canRead: [userOwns, "sunshineRegiment", "admins"],
+    canCreate: ["members"],
+    canUpdate: [userOwns, "sunshineRegiment", "admins"],
+  },
+  /** The user voting */
+  userId: {
+    ...foreignKeyField({
+      idFieldName: "userId",
+      resolverName: "user",
+      collectionName: "Users",
+      type: "User",
+      nullable: false,
+    }),
+    canRead: [userOwns, "sunshineRegiment", "admins"],
+    canCreate: ["members"],
+    canUpdate: [userOwns, "sunshineRegiment", "admins"],
+  },
+  /**
+   * Object like {'cF8iwCmwFjbmCqYkQ': 1, 'cF8iwCmwFjbmCqYkQ': 2} representing the (unnormalised) weights
+   * given to each selected candidate. Weights can not be negative, but can be null (these will
+   * be ignored).
+   */
+  vote: {
+    type: Object,
```

Comment:
This is not typed (in the db, or in the returned object), for a number of reasons:
- the `Record<string, number | null>` type is hard (not possible maybe) to represent as a `SimpleSchema` object
- returning a json blob typed as a specific object in graphql actually requires a ton of boilerplate (see e.g. `socialPreview`), and I thought it was less complex overall in this case to just not have the type safety (as we know this will only be used in a few places)